### PR TITLE
load_data.py Refinement Following Review Feedback

### PR DESCRIPTION
**Pull request by RongLirr**
- Removed random windowing of fluent_clip, replacing it with a fixed-length slice from the start of the sequence (fluent_clip = data[:fluent_frames]). → to avoid the one-to-many mapping issue between identical conditions and differing targets, as highlighted by Mingyi, which may otherwise hinder stable learning.

- Replaced in-place normalization with per-sample standardization using (x - mean) / std.

- Removed input_mask from the output dictionary, as it was not used in the loss function or model forward pass.

- Added explicit padding to fluent_clip using the last valid frame when sequence length is shorter than fluent_frames. → for a consistent target length across samples, which is critical when applying MSE loss directly without masking (for now)

- Maintained the use of zero_pad_collator, limited to the disfluent_seq input.

### CODE REVIEW

**Comment by RongLirr** [comment]
Apply dual-stage normalization (shoulder-based + global mean/std) with per-sample pose loading

**Comment by AmitMY** [comment]
Seems to be missing the anonymization dependency in the pyproject
also seems to be not un-normalizing after prediction etc

**Comment by RongLirr** [comment]
I followed your advice and switched to using normalize_mean_std from the pose-anonymization library, which should apply the full, precomputed global mean and standard deviation uniformly to every clip. However, my debug outputs reveal that the post-normalization statistics still vary wildly from sample to sample:

Reasonable case:

```

[After Norm] Fluent data mean: 0.34, std: 1.53  
[After Norm] Disfluent data mean: 0.10, std: 1.09
```

Problematic cases:

```

[After Norm] Fluent data mean: 8.26, std: 44.84  
[After Norm] Disfluent data mean: 8.26, std: 44.84

or

[After Norm] Fluent data mean: 4.53, std: 33.96  
[After Norm] Disfluent data mean: 1.29, std: 19.91
```



Because of this inconsistency (some clips end up with a normalized standard deviation of ~1, while others exceed 40) my training loss starts at an extremely high value (~350) and slowly decreases to 1 over 10000 epochs.

By contrast, in Mingyi’s implementation he computes mean and std over the entire dataset in one go and then applies those statistics uniformly. That approach immediately yields a consistently small initial loss (<1). However, it requires loading all data at once (into a list) rather than streaming samples dynamically.

Would you recommend switching to that dataset-wide, mean/std calculation as a more reliable way to enforce truly uniform normalization? Or is there another issue in how I’m currently applying the precomputed pose-anonymization stats that I should be aware of?

Thanks for any further guidance!

P.S. here is an example debug printing info for epoch 5:

```
[DEBUG] noise generated mean: 0.0006062586326152086, std: 1.0008467435836792
[DEBUG] x_t after q_sample shape: torch.Size([160, 178, 3, 40]), mean: 0.8452344536781311, std: 11.616196632385254
[DEBUG] model_output shape: torch.Size([160, 178, 3, 40]), mean: 0.007164412643760443, std: 0.07425284385681152
[DEBUG] target shape: torch.Size([160, 178, 3, 40]), mean: 1.6851012706756592, std: 18.825048446655273
[DEBUG] frame_mask True count: 6400, False count: 0
[DEBUG] loss_data: 357.0606994628906
[DEBUG][Before Norm] Fluent raw data mean: 244.25341796875, std: 66.96237182617188
[DEBUG][Before Norm] Disfluent raw data mean: 247.79910278320312, std: 99.73554229736328
[DEBUG][After Norm] Fluent data mean: 4.433347225189209, std: 30.858699798583984
[DEBUG][After Norm] Disfluent data mean: 1.7004530429840088, std: 19.852664947509766
[DEBUG][Before Norm] Fluent raw data mean: 206.6322021484375, std: 93.5718765258789
[DEBUG][Before Norm] Disfluent raw data mean: 206.6322021484375, std: 93.5718765258789
[DEBUG][After Norm] Fluent data mean: 8.266307830810547, std: 44.83652877807617
[DEBUG][After Norm] Disfluent data mean: 8.266307830810547, std: 44.83652877807617
[DEBUG][Before Norm] Fluent raw data mean: 230.61265563964844, std: 57.9368896484375
[DEBUG][Before Norm] Disfluent raw data mean: 228.4539337158203, std: 74.18541717529297
[DEBUG][After Norm] Fluent data mean: 0.20718121528625488, std: 1.1208118200302124
[DEBUG][After Norm] Disfluent data mean: 22.095060348510742, std: 72.36400604248047
...
Epoch 5/10000 | loss: 357.017914 | best_loss: 357.017914:   0%|          | 5/10000 [00:23<11:03:10,  3.98s/it]INFO:Logger:Epoch 5/10000 | loss_data: 357.017914, loss: 357.017914,  | best_loss: 357.017914
Epoch 5/10000 | loss: 357.017914 | best_loss: 357.017914:   0%|          | 6/10000 [00:24<10:57:47,  3.95s/it][DEBUG] x_start permuted shape: torch.Size([160, 178, 3, 40]), mean: 1.6851012706756592, std: 18.825048446655273
```
